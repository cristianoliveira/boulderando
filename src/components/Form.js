import { useState } from 'react'
import styled from '@emotion/styled'

import FormControl from '@mui/material/FormControl'
import FormHelperText from '@mui/material/FormHelperText'
import Button from '@mui/material/Button'
import Grid from '@mui/material/Grid'
import { useForm } from 'react-hook-form'
import Alert from '@mui/material/Alert'
import AlertTitle from '@mui/material/AlertTitle'

import TextField from './TextField'
import DateField from './DateField'

const StyledForm = styled.form`
  padding: 20px;
`

const FULL_GRID = 12

const showErrorFields = errors => Object.keys(errors).map(e =>e).join(', ')

function Form({ onSubmit }) {
  const {
    register,
    handleSubmit,
    formState: { errors, isValid, isSubmitted },
  } = useForm()

  return (
    <StyledForm onSubmit={handleSubmit(onSubmit)}>
      <Grid
        direction="row"
        justify="flex-start"
        alignItems="flex-start"
        spacing={2}
        container
      >
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            name="name"
            label="First Name"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField name="last_name" label="Last Name" register={register} />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <DateField name="birthday" label="Birthday" register={register} />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField
            name="address"
            label="Street address"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            name="postal_code"
            label="Postal Code"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            name="city"
            label="City"
            value="Berlin"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField name="email" label="Email" register={register} />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField
            name="usc_number"
            label="Urban Sports Club No."
            register={register}
          />
          <FormHelperText>Your USB member number.</FormHelperText>
        </Grid>
        <Grid item xs={FULL_GRID}>
          {isSubmitted && !isValid && (
            <Alert severity="error">All fields are required. Missing fields: {showErrorFields(errors)}</Alert>
          )}
        </Grid>
      </Grid>
      <FormControl fullWidth>
        <Button variant="contained" type="submit">
          Save data
        </Button>
      </FormControl>
      <FormHelperText variant="outlined">
        ** This data is stored in your browser and can be deleted at any time
      </FormHelperText>
    </StyledForm>
  )
}

export default Form
