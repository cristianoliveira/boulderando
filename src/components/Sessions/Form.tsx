import styled from '@emotion/styled'

import FormControl from '@mui/material/FormControl'
import FormHelperText from '@mui/material/FormHelperText'
import Button from '@mui/material/Button'
import Grid from '@mui/material/Grid'
import { useForm } from 'react-hook-form'
import Alert from '@mui/material/Alert'

import SelectField from '../fields/SelectField'

import * as TID from '../../constants/data-testid'

const StyledForm = styled.form`
  padding: 20px;
`

const FULL_GRID = 12

const showErrorFields = (errors: object): string =>
  Object.keys(errors)
    .map((e) => e)
    .join(', ')

// Sometime in life you have to just have faith
// range from 8:00 to 21:30
const timesSelection: Array<string> = [...Array(20).keys()]
  .filter((i) => i >= 8 && i <= 20)
  .map((i) => [`${i}:00 - ${i + 2}:00`, `${i}:30 - ${i + 2}:30`])
  .flat()

function Form({ onSubmit }: Submitable): JSX.Element {
  const {
    register,
    handleSubmit,
    formState: { errors, isValid, isSubmitted },
  } = useForm({})

  const errorsList = showErrorFields(errors)
  return (
    <StyledForm
      onSubmit={handleSubmit((data) =>
        onSubmit({ ...data, day_of_week: data.human_date.replace(/this /, '') })
      )}
    >
      <Grid
        direction="row"
        alignItems="flex-start"
        spacing={2}
        container
      >
        <Grid item xs={FULL_GRID}>
          <SelectField
            data-testid={TID.CUSTOM_SESSION_FORM_GYM}
            name="gym_name"
            label="Gym"
            register={register}
            selectOptions={[
              { label: 'Boulderklub', value: 'boulderklub' },
              { label: 'Basement', value: 'basement' },
            ]}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <SelectField
            data-testid={TID.CUSTOM_SESSION_FORM_DAY}
            name="human_date"
            label="Day of the week"
            register={register}
            selectOptions={[
              { label: 'Monday', value: 'this monday' },
              { label: 'Tuesday', value: 'this tuesday' },
              { label: 'Wednesday', value: 'this wednesday' },
              { label: 'Thursday', value: 'this thursday' },
              { label: 'Friday', value: 'this friday' },
              { label: 'Saturday', value: 'this saturday' },
            ]}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <SelectField
            data-testid={TID.CUSTOM_SESSION_FORM_TIME}
            name="time"
            label="Time"
            register={register}
            selectOptions={timesSelection.map((time) => ({
              label: time,
              value: time,
            }))}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          {isSubmitted && !isValid && errorsList && (
            <Alert
              data-testid={TID.USER_FORM_ERROR_MESSAGE_CONTAINER}
              severity="error"
            >
              All fields are required. Missing fields: {errorsList}
            </Alert>
          )}
        </Grid>
      </Grid>
      <FormControl fullWidth>
        <Button
          data-testid={TID.CUSTOM_SESSION_FORM_SUBMIT_BUTTON}
          variant="contained"
          type="submit"
        >
          Save data
        </Button>
      </FormControl>
      <FormHelperText variant="outlined">
        ** This data is stored in your browser and can be deleted at any time
      </FormHelperText>
    </StyledForm>
  )
}

export default Form
