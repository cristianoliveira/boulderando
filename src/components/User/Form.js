import { useState } from 'react'
import styled from '@emotion/styled'

import FormControl from '@mui/material/FormControl'
import FormHelperText from '@mui/material/FormHelperText'
import Button from '@mui/material/Button'
import Grid from '@mui/material/Grid'
import { useForm } from 'react-hook-form'
import Alert from '@mui/material/Alert'
import AlertTitle from '@mui/material/AlertTitle'

import PhoneField from '../fields/PhoneField'
import TextField from '../fields/TextField'
import DateField from '../fields/DateField'

import * as TID from '../../constants/data-testid'

const StyledForm = styled.form`
  padding: 20px;
`

const FULL_GRID = 12

const showErrorFields = (errors) =>
  Object.keys(errors)
    .map((e) => e)
    .join(', ')

function Form({ user = {}, onSubmit }) {
  const {
    register,
    handleSubmit,
    formState: { errors, isValid, isSubmitted },
  } = useForm({
    defaultValues: { type: 'Urban Sports Club', ...user },
  })

  const errorsList = showErrorFields(errors)
  return (
    <StyledForm onSubmit={handleSubmit(onSubmit)}>
      <Grid
        direction="row"
        justify="flex-start"
        alignItems="flex-start"
        spacing={2}
        container
      >
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_FIRST_NAME}
            name="name"
            label="First Name"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_LAST_NAME}
            name="last_name"
            label="Last Name"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <DateField
            data-testid={TID.USER_INPUT_BIRTHDAY}
            name="birthday"
            label="Birthday"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField
            data-testid={TID.USER_INPUT_STREET_ADDRESS}
            name="address"
            label="Street address"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_POSTALCODE}
            name="postal_code"
            label="Postal Code"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_CITY}
            name="city"
            label="City"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField
            data-testid={TID.USER_INPUT_EMAIL}
            name="email"
            label="Email"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <PhoneField
            data-testid={TID.USER_INPUT_PHONE_NUMBER}
            name="phone_number"
            label="Phone Number"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <div style={{ visibility: 'hidden', height: 0 }}>
            <TextField name="type" register={register} />
          </div>
          <TextField
            data-testid={TID.USER_INPUT_URBAN_SPORT_NUMBER}
            name="usc_number"
            label="Urban Sports Club No."
            register={register}
          />
          <FormHelperText>Your USB member number.</FormHelperText>
        </Grid>
        <Grid item xs={FULL_GRID}>
          {isSubmitted && !isValid && errorsList && (
            <Alert
            data-testid={TID.USER_FORM_ERROR_MESSAGE_CONTAINER}
            severity="error">
              All fields are required. Missing fields: {errorsList}
            </Alert>
          )}
        </Grid>
      </Grid>
      <FormControl fullWidth>
        <Button
          data-testid={TID.USER_FORM_SUBMIT_BUTTON}
          variant="contained"
          type="submit"
        >
          Save data
        </Button>
      </FormControl>
      <FormHelperText variant="outlined">
        ** This data is stored in your browser and can be deleted at any time
      </FormHelperText>
    </StyledForm>
  )
}

export default Form
