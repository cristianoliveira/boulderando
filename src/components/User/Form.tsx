import styled from '@emotion/styled'

import { useForm, FieldError } from 'react-hook-form'

import FormControl from '@mui/material/FormControl'
import FormHelperText from '@mui/material/FormHelperText'
import Button from '@mui/material/Button'
import Grid from '@mui/material/Grid'
import Alert from '@mui/material/Alert'

import PhoneField from '../fields/PhoneField'
import TextField from '../fields/TextField'
import DateField from '../fields/DateField'

import * as TID from '../../constants/data-testid'

const StyledForm = styled.form`
  padding: 20px;
`

const FULL_GRID = 12

type UserFormProps = {
  user?: User
  telegramIdParam: string
} & Submitable

const showErrorFields = (errors: Partial<FieldError>) =>
  Object.keys(errors)
    .map((e) => e)
    .join(', ')

function Form({ user, telegramIdParam, onSubmit }: UserFormProps) {
  const {
    register,
    handleSubmit,
    setValue,
    formState: { errors, isValid, isSubmitted },
  } = useForm({
    defaultValues: {
      ...user,
      type: 'Urban Sports Club',
    },
  })

  setValue('telegram_id', telegramIdParam || user?.telegram_id)

  const errorsList = showErrorFields(errors as Partial<FieldError>)
  return (
    <StyledForm
      onSubmit={handleSubmit((data) => {
        onSubmit(data)
      })}
    >
      <Grid
        direction="row"
        alignItems="flex-start"
        spacing={2}
        container
      >
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_FIRST_NAME}
            name="name"
            label="First Name"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_LAST_NAME}
            name="last_name"
            label="Last Name"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <DateField
            data-testid={TID.USER_INPUT_BIRTHDAY}
            name="birthday"
            label="Birthday"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField
            data-testid={TID.USER_INPUT_STREET_ADDRESS}
            name="address"
            label="Street address"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_POSTALCODE}
            name="postal_code"
            label="Postal Code"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_CITY}
            name="city"
            label="City"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField
            data-testid={TID.USER_INPUT_EMAIL}
            name="email"
            label="Email"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <PhoneField
            data-testid={TID.USER_INPUT_PHONE_NUMBER}
            name="phone_number"
            label="Phone Number"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <div style={{ visibility: 'hidden', height: 0 }}>
            <TextField data-testid="type" name="type" register={register} />
          </div>
          <TextField
            data-testid={TID.USER_INPUT_URBAN_SPORT_NUMBER}
            name="usc_number"
            label="Urban Sports Club No."
            register={register}
          />
          <FormHelperText>Your USB member number.</FormHelperText>
        </Grid>
        <Grid item xs={FULL_GRID}>
          {isSubmitted && !isValid && errorsList && (
            <Alert
              data-testid={TID.USER_FORM_ERROR_MESSAGE_CONTAINER}
              severity="error"
            >
              All fields are required. Missing fields: {errorsList}
            </Alert>
          )}
        </Grid>
      </Grid>
      <FormControl fullWidth>
        <Button
          data-testid={TID.USER_FORM_SUBMIT_BUTTON}
          variant="contained"
          type="submit"
        >
          Save data
        </Button>
      </FormControl>
      <FormHelperText variant="outlined">
        Your personal data is stored only in your browser and can be deleted at
        any time. When scheduling specific sessions, your data is transmitted to
        the corresponding Gym using the Dr. Plano booking system. By scheduling
        sessions you&apos;re automatically agreeing to the Terms and Conditions
        and Privacy Policy of that Gym.
      </FormHelperText>
    </StyledForm>
  )
}

export default Form
