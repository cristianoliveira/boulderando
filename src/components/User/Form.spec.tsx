import React from 'react'
import { act } from '@testing-library/react'
// eslint-disable-next-line
import renderer from 'react-test-renderer'

import Form from './Form'

const user = {
  name: 'John',
  last_name: 'Doe',
  birthday: '1969-04-20',
  address: 'Mein Str, 22b',
  postal_code: '00000',
  city: 'Berlin',
  phone_number: '000000000000',
  email: 'my@gmail.com',
  usc_number: '000000000',
  type: 'Urban Sports Club',
  telegram_id: '111111',
}

describe('Form', () => {
  it('matches the snapshot', async () => {
    expect(
      renderer
        .create(
          <Form user={user} telegramIdParam="123123" onSubmit={jest.fn()} />
        )
        .toJSON()
    ).toMatchSnapshot()
  })
})
