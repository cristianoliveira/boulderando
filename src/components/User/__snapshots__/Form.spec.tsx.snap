// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches the snapshot 1`] = `
<form
  className="css-gz8dae"
  onSubmit={[Function]}
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-lndw90-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth css-zevgu-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="name"
        >
          First Name
        </label>
        <div
          className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
          data-testid="user--first_name--input"
          onClick={[Function]}
        >
          <input
            aria-describedby="Input for name"
            aria-invalid={false}
            className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id="name"
            name="name"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth css-zevgu-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="last_name"
        >
          Last Name
        </label>
        <div
          className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
          data-testid="user--last_name--input"
          onClick={[Function]}
        >
          <input
            aria-describedby="Input for last_name"
            aria-invalid={false}
            className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id="last_name"
            name="last_name"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth css-mo9nhr-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="birthday"
        >
          Birthday
        </label>
        <div
          className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
          data-testid="user--birthday--input"
          onClick={[Function]}
        >
          <input
            aria-describedby="Input for birthday"
            aria-invalid={false}
            className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id="birthday"
            name="birthday"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="date"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth css-zevgu-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="address"
        >
          Street address
        </label>
        <div
          className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
          data-testid="user--street-address--input"
          onClick={[Function]}
        >
          <input
            aria-describedby="Input for address"
            aria-invalid={false}
            className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id="address"
            name="address"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth css-zevgu-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="postal_code"
        >
          Postal Code
        </label>
        <div
          className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
          data-testid="user--postal_code--input"
          onClick={[Function]}
        >
          <input
            aria-describedby="Input for postal_code"
            aria-invalid={false}
            className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id="postal_code"
            name="postal_code"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth css-zevgu-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="city"
        >
          City
        </label>
        <div
          className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
          data-testid="user--city--input"
          onClick={[Function]}
        >
          <input
            aria-describedby="Input for city"
            aria-invalid={false}
            className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id="city"
            name="city"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth css-zevgu-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="email"
        >
          Email
        </label>
        <div
          className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
          data-testid="user--input--email"
          onClick={[Function]}
        >
          <input
            aria-describedby="Input for email"
            aria-invalid={false}
            className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id="email"
            name="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth css-zevgu-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="phone_number"
        >
          Phone Number
        </label>
        <div
          className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
          data-testid="user--input_phone--number"
          onClick={[Function]}
        >
          <input
            aria-describedby="Input for phone_number"
            aria-invalid={false}
            className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id="phone_number"
            name="phone_number"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="0000 00000000"
            required={false}
            type="tel"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        style={
          Object {
            "height": 0,
            "visibility": "hidden",
          }
        }
      >
        <div
          className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth css-zevgu-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="type"
          />
          <div
            className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
            data-testid="type"
            onClick={[Function]}
          >
            <input
              aria-describedby="Input for type"
              aria-invalid={false}
              className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              disabled={false}
              id="type"
              name="type"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth css-zevgu-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="usc_number"
        >
          Urban Sports Club No.
        </label>
        <div
          className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
          data-testid="user--input--urban_sport_number"
          onClick={[Function]}
        >
          <input
            aria-describedby="Input for usc_number"
            aria-invalid={false}
            className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id="usc_number"
            name="usc_number"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
      </div>
      <p
        className="MuiFormHelperText-root css-1d1r5q-MuiFormHelperText-root"
      >
        Your USB member number.
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    />
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      data-testid="user--form_submit--button"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Save data
    </button>
  </div>
  <p
    className="MuiFormHelperText-root MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
  >
    Your personal data is stored only in your browser and can be deleted at any time. When scheduling specific sessions, your data is transmitted to the corresponding Gym using the Dr. Plano booking system. By scheduling sessions you're automatically agreeing to the Terms and Conditions and Privacy Policy of that Gym.
  </p>
</form>
`;
